pkg/manifests/manifests.go

```diff
@@ -142,8 +169,52 @@ func (f *Factory) RouterDeployment(cr *ingressv1alpha1.ClusterIngress) (*appsv1.
 	}
 	deployment.Spec.Selector.MatchLabels["router"] = name
 
+	statsSecretName := fmt.Sprintf("router-stats-%s", cr.Name)
 	env := []corev1.EnvVar{
 		{Name: "ROUTER_SERVICE_NAME", Value: cr.Name},
+		{Name: "STATS_USERNAME", ValueFrom: &corev1.EnvVarSource{
+			SecretKeyRef: &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: statsSecretName,
+				},
+				Key: "statsUsername",
+			},
+		}},
+		{Name: "STATS_PASSWORD", ValueFrom: &corev1.EnvVarSource{
+			SecretKeyRef: &corev1.SecretKeySelector{
+				LocalObjectReference: corev1.LocalObjectReference{
+					Name: statsSecretName,
+				},
+				Key: "statsPassword",
+			},
+		}},
+	}
+
+	if enableMetrics {
```

Comment:
We always want metrics, this can all be inlined above and `enableMetrics` can be removed